src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,208 @@
+
+This module defines a credential plugin for GitHub Apps, allowing
+authentication via GitHub App tokens.
+
+Functions:
+- github_app_backend: Generates a GitHub App token fields from GitHub Admin.
+- github_app_lookup: Defines the credential plugin interface.
+"""
+# FIXME: ManagedCredentialType type not found.
+
+from typing import TypedDict
+
+from awx_plugins.interfaces._temporary_private_api import (  # noqa: WPS436
+    ManagedCredentialType,
+)
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+from github import Auth, Github
+
+from .injectors import github_app_injector
+from .plugin import CredentialPlugin
+
+
+# Define expected structure for 'fields'
+class FieldDict(TypedDict, total=False):
+    id: str
+    label: str
+    type: str
+    format: str  # Optional field
+    secret: bool  # Optional field
+    multiline: bool  # Optional field
+    help_text: str
+    default: str
+
+# Define expected structure for 'metadata'
+
+
+class MetadataDict(TypedDict, total=False):
+    id: str
+    label: str
+    type: str
+    help_text: str
+
+# Define the full structure of github_app_inputs
+
+
+class GitHubAppInputs(TypedDict):
+    fields: list[FieldDict]
+    metadata: list[MetadataDict]
+    required: list[str]
+
+
+# Explicitly define the type of github_app_inputs
+github_app_inputs: GitHubAppInputs = {
+    'fields': [
+        {
+            'id': 'github_url',
+            'label': _('GitHub URL'),
+            'type': 'string',
+            'help_text': _(
+                'Specify the GitHub URL here in the case of an Enterprise Github. '
+                'Example: https://example.github.com',
+            ),
+            'default': 'https://github.com',
+        },
+        {
+            'id': 'app_id',
+            'label': _('Github App id created by the GitHub Admin'),
+            'type': 'string',
+            'help_text': _(
+                'The GitHub App Id created by the GitHub Admin. '
+                'Example App ID: 1121547 '
+                'from https://github.com/settings/apps/arestlel-github-app-ansible '
+                'required for creating a bearer token for authentication',
+            ),
+            'default': '1121547',
+        },
+        {
+            'id': 'install_id',
+            'label': _('Installation id'),
+            'type': 'string',
+            'help_text': _(
+                'The Installation ID from the GitHub App installation generated by the GitHub Admin. '
+                'Example: 59980338 '
+                'extracted from the installation link: https://github.com/settings/installations/59980338 '
+                'required for creating a limited GitHub app token',
+            ),
+            'default': '59980338',
+        },
+        {
+            'id': 'ssh_key_data',
+            'label': _('RSA Private Key'),
+            'type': 'string',
+            'format': 'ssh_private_key',
```

Comment:
This might be fine but would need an explainer comment
```suggestion
            'format': 'ssh_private_key',  # this validator checks for formats close to PKCS#1 PEMs
```